check_pie_supported seems to run earlier so the run above returns nothing and this fails
--
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,7 +125,7 @@
   OUTPUT_VARIABLE ZIG_PIE_SUPPORTED_BY_CMAKE
   LANGUAGES C CXX
 )
-if(ZIG_PIE AND NOT ZIG_PIE_SUPPORTED_BY_CMAKE)
+if(0 AND ZIG_PIE AND NOT ZIG_PIE_SUPPORTED_BY_CMAKE)
   message(SEND_ERROR "ZIG_PIE was requested but CMake does not support it for \"zigcpp\" target")
 endif()
 
